rxandroidble/src/main/java/com/polidea/rxandroidble/internal/util/MismatchDataHandler.java

```diff
@@ -0,0 +1,10 @@
+package com.polidea.rxandroidble.internal.util;
+
+
+/**
+ * Handler for {@link com.polidea.rxandroidble.internal.util.IllegalOperationChecker.MismatchData} returned by
+ * {@link IllegalOperationChecker}.
+ */
+public interface MismatchDataHandler {
```

Comment:
Why not just a `IllegalOperationHandler`? And it can be put just in `internal.connection` package. Does not look like an util that is used all over the code.